### PR TITLE
fix Toast stacking

### LINKED ISSUE #367: Toaster create method should not have the `inline` property hard coded
### Bug report

- __Package version(s)__: 1.3.1
- __Browser and OS versions__: all

#### Steps to reproduce

Create a custom Toaster with `inline` set to `false`, e.g., using code from the docs:

```
import { Position, Toaster } from "@blueprintjs/core";
 
export const OurToaster = Toaster.create({
    className: "my-toaster",
    position: Position.BOTTOM_RIGHT,
    inline: false
});
```

#### Actual behavior

Observe the `pt-overlay-inline` class in the generated DOM element:

`<span data-reactroot="" class="pt-overlay pt-overlay-open pt-overlay-inline pt-toast-container pt-toast-container-top">...</span>`

#### Expected behavior

The create method that returns a new Toaster instance should not have the `inline` property hard coded:
https://github.com/palantir/blueprint/blob/release-1.3.1/packages/core/src/components/toast/toaster.tsx#L94

### PR DESCRIPTION
**Pull request by giladgray**
#### Checklist

- [x] Include tests
- [x] Update documentation

#### Changes proposed in this pull request:

- clone Toast props before mutating. this avoids leaking the mutation and allows more than one of each Toast to appear in the example.
- ensure Toasts are always `position: relative` so they also stack when `inline`.
- warn when passing inline prop to Toaster.create()

#### Reviewers should focus on:

- are you into this new warning or is it overkill? i updated the props docs with the same content.

### CODE REVIEW

File: packages/core/src/components/toast/toaster.tsx [hunk 1]
```diff
@@ -89,6 +93,9 @@ export class Toaster extends AbstractComponent<IToasterProps, IToasterState> imp
      * The `Toaster` will be rendered into a new element appended to the given container.
      */
     public static create(props?: IToasterProps, container = document.body): IToaster {
+        if (props != null && props.inline != null) {
```
Comment (adidahiya) [inline packages/core/src/components/toast/toaster.tsx]: would prefer explicit checks against `undefined`; we don't really need to support nulls

Comment (giladgray) [inline packages/core/src/components/toast/toaster.tsx]: i'm just so used to doing it this way. 

how about if we tackle that change across the codebase when we enable `strictNullChecks`?

Comment (adidahiya) [inline packages/core/src/components/toast/toaster.tsx]: sure, but it won't get flagged for you automatically; you'll need to look for all such comparisons against `null`. in my ideal world, we also enable the `no-null-keyword` lint rule and any explicit usage of `null` would have to be whitelisted

Comment (giladgray) [inline packages/core/src/components/toast/toaster.tsx]: sure that sounds reasonable!

**Pull request merged**